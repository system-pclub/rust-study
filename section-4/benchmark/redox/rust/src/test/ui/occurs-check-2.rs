#![feature(box_syntax)]

fn main() {
    let f;
    let g;
    g = f;
    f = box g;
    //~^  ERROR mismatched types
    //~| cyclic type of infinite size
}
