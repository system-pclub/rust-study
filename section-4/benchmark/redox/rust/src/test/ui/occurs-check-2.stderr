error[E0308]: mismatched types
  --> $DIR/occurs-check-2.rs:7:9
   |
LL |     f = box g;
   |         ^^^^^
   |         |
   |         cyclic type of infinite size
   |         help: try using a conversion method: `(box g).to_string()`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
